src/main/java/uk/gov/pay/ledger/transaction/model/PaymentFactory.java

```diff
@@ -0,0 +1,84 @@
+
+public class PaymentFactory {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(PaymentFactory.class);
+    private ObjectMapper objectMapper;
+
+    @Inject
+    public PaymentFactory(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public Payment fromTransactionEntity(TransactionEntity entity) {
+        try {
+            JsonNode transactionDetails = objectMapper.readTree(Optional.ofNullable(entity.getTransactionDetails()).orElse("{}"));
+            Address billingAddress = Address.from(
+                    safeGetAsString(transactionDetails, "address_line1"),
+                    safeGetAsString(transactionDetails, "address_line2"),
+                    safeGetAsString(transactionDetails, "address_postcode"),
+                    safeGetAsString(transactionDetails, "address_city"),
+                    safeGetAsString(transactionDetails, "address_county"),
+                    safeGetAsString(transactionDetails, "address_country")
+            );
+
+            CardDetails cardDetails = CardDetails.from(entity.getCardholderName(), billingAddress, entity.getCardBrand(),
+                    entity.getLastDigitsCardNumber(), entity.getFirstDigitsCardNumber(),
+                    safeGetAsString(transactionDetails, "card_expiry_date"));
+
+            Map<String, Object> metadata = null;
+            if(entity.getExternalMetadata() != null) {
+                metadata = objectMapper.readValue(entity.getExternalMetadata(), new TypeReference<Map<String, Object>>() {});
+            }
+
+            RefundSummary refundSummary = RefundSummary.ofValue(entity.getRefundStatus(), entity.getRefundAmountAvailable(), entity.getRefundAmountSubmitted());
+            SettlementSummary settlementSummary = SettlementSummary.ofValue(entity.getSettlementSubmittedTime(), entity.getSettledTime());
+
+            return new Payment(entity.getGatewayAccountId(), entity.getAmount(), entity.getReference(), entity.getDescription(),
+                    TransactionState.from(entity.getState()), safeGetAsString(transactionDetails, "language"),
+                    entity.getExternalId(), safeGetAsString(transactionDetails, "return_url"), entity.getEmail(),
+                    safeGetAsString(transactionDetails, "payment_provider"), entity.getCreatedDate(),
+                    cardDetails, Boolean.valueOf(safeGetAsString(transactionDetails, "delayed_capture")),
+                    metadata, entity.getEventCount(), safeGetAsString(transactionDetails, "gateway_transaction_id"),
+                    safeGetAsLong(transactionDetails, "corporate_surcharge"), safeGetAsLong(transactionDetails, "fee"),
+                    entity.getNetAmount(), refundSummary, entity.getTotalAmount(), settlementSummary);
+        } catch (IOException e) {
+            LOGGER.error(format("Error during the parsing transaction entity data {}", entity.getExternalId()), e);
+            throw new RuntimeException(e);
```

Comment:
Is there something that catches and deals with this exception? It might be outside of this PR but if there isn't it might not need to be rethrown